pkgs/development/compilers/koka/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, cmake, makeWrapper, mkDerivation, fetchFromGitHub
+, alex, array, base, bytestring, cond, containers, directory, extra
+, filepath, haskeline, hpack, hspec, hspec-core, json, lib, mtl
+, parsec, process, regex-compat, text, time }:
+
+let
+  version = "2.1.1";
+  src = fetchFromGitHub {
+    owner = "koka-lang";
+    repo = "koka";
+    rev = "v${version}";
+    sha256 = "sha256-cq+dljfTKJh5NgwQfxQQP9jRcg2PQxxBVEgQ59ll36o=";
+    fetchSubmodules = true;
+  };
+  kklib = stdenv.mkDerivation {
+    pname = "kklib";
+    inherit version;
+    src = "${src}/kklib";
+    nativeBuildInputs = [ cmake ];
+  };
+  runtimeDeps = [ cmake stdenv.cc ];
+in
+mkDerivation rec {
+  pname = "koka";
+  inherit version src;
+  isLibrary = false;
+  isExecutable = true;
+  libraryToolDepends = [ hpack ];
+  executableHaskellDepends = [
+    array base bytestring cond containers directory haskeline mtl
+    parsec process text time kklib
+  ];
+  executableToolDepends = [ alex makeWrapper ];
+  postInstall = ''
+    mkdir -p $out/share/koka/v${version}
+    cp -a lib $out/share/koka/v${version}
+    cp -a contrib $out/share/koka/v${version}
+    cp -a kklib $out/share/koka/v${version}
+    wrapProgram "$out/bin/koka" \
```

Comment:
Okay, so `stdenv.cc` is wrong, since it isn't suitable for cross, my bad. We want a C compiler for the buildPlatform and not take the C compiler of the hostPlatform.

Also for koka to properly pick up on the C compiler, we need to `--set CC` in `wrapProgram` which is actually kind of neat.

OTOH statically setting a C compiler may not the best idea since users may want to choose different ones, for example by setting `CC`?? Easiest would probably to statically take `gcc` as an input, I'm not sure.

Reply:
I could set it to buildPackages.stdenv.cc`

Reply:
Seems overrideable enough, would be especially neat if we had `pkgsClang` :|